File: backend/src/main/java/com/hpms/backend/model/Housekeeping.java
package com.hpms.backend.model;

import com.hpms.backend.enumCollection.HousekeepingPriorityEnum;
import com.hpms.backend.enumCollection.HousekeepingStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "housekeeping")
public class Housekeeping {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = true)
    @JoinColumn(name = "employee_id", referencedColumnName = "id")
    private User user;

    @ManyToOne(optional = false)
    @JoinColumn(name = "room_id", referencedColumnName = "id")
    private Room room;

    @Column(columnDefinition = "TEXT")
    private String note;

    @Enumerated(EnumType.STRING)
    private HousekeepingStatus status;

    @Enumerated(EnumType.STRING)
    private HousekeepingPriorityEnum priority = HousekeepingPriorityEnum.LOW;

    private LocalDate assignedDate;
    private LocalDate completionDate;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/interfaces/housekeeping.ts
import type { FormValues } from "../lib/form";
import type { HousekeepingPriority, HousekeepingStatus } from "./enums";
import type { RoomData } from "./room";
import type { UserData } from "./user";

export interface HousekeepingData {
	id: number;
	user?: UserData;
	room: RoomData;
	note?: string;
	status: HousekeepingStatus;
	priority: HousekeepingPriority;
	assignedDate?: Date;
	completionDate?: Date;
	createdAt: Date;
	updatedAt?: Date;
}

export interface HousekeepingFormData extends FormValues {
	userId?: number;
	roomId: number;
	note: string;
	priority: HousekeepingPriority;
}

export interface CreateHousekeepingPayload {
	userId?: number;
	roomId: number;
	note: string;
	priority: HousekeepingPriority;
}

export interface UpdateHousekeepingPayload {
	userId?: number;
	roomId: number;
	note: string;
	priority: HousekeepingPriority;
}
